tools/unsupervised_dataset/sql_crawler/extraction_modules/generic_extraction_module.py

```diff
@@ -1,8 +1,32 @@
-class GenericExtractionModule(object):
+""" Collects SQL queries from any website """
+import html
+import re
+
+class GenericExtractionModule():
     """ A module to extract SQL queries from any website without knowing the HTML
         format of a site.
     """
 
-    def find_queries(html):
-        #TODO(Noah): Implement generic module
-        return []
+    def find_queries(html_response):
+        """ Finds queries and extracts them from any website, without using HTML
+        tags to locate them.
+
+        Args:
+            html: HTML response which contains HTML text
+
+        Returns
+            A list of queries in the form of strings.
+        """
+        # Remove HTML tags and special characters
+        content = html_response.text
+        converted = html.unescape(content)
+        tags_removed = re.sub('<[^<]+?>', '', converted)
+
+        # Look for text that matches common SQL queries
+        matches = re.findall(r"(?:SELECT|WITH|CREATE)[^;]*;", tags_removed)
```

Comment:
Maybe define this as a separate const string. I think this can be longer in the future when you cover insert, update, merge, etc.

Reply:
+1